src/udev/udev-varlink.c

```diff
@@ -113,42 +114,68 @@ static int vl_method_exit(sd_varlink *link, sd_json_variant *parameters, sd_varl
         return sd_varlink_reply(link, NULL);
 }
 
+int manager_init_varlink_server(Manager *manager, int fd) {
+        int r;
+
+        assert(manager);
+
+        /* This takes passed file descriptor on success. */
+
+        if (fd >= 0) {
+                if (manager->varlink_server)
+                        return log_warning_errno(SYNTHETIC_ERRNO(EALREADY), "Received multiple varlink socket (%i), ignoring.", fd);
```

Comment:
tbh i would just have allowed multiple varlink sockets to be passed, and would just use them all. i think this is utlimately a pretty powerful concept, because it allows people to bind additional sockets inside of chroots and stuff and things would just work. sd-varlink after all doesn't really care how many listener sockets and how many connections you set it up with.